src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import java.time.LocalDate;
+
+public class Parser {
+    private String command;
+    private Storage storage;
+    private TaskList taskList;
+    private boolean isEnd = false;
+
+    public Parser(String command, Storage storage, TaskList taskList) {
+        this.command = command;
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+    public void parse() {
+        try{
+            if (command.startsWith("list")) {
+                this.taskList.listTasks();
+            } else if (command.startsWith("mark")) {
+                this.taskList.markTask(Integer.valueOf(command.split(" ")[1]) - 1);
+            } else if (command.startsWith("todo")) {
+                if (command.split(" ", 2).length == 1) {
+                    throw new DukeException(" OOPS!!! The description of a todo cannot be empty.");
+                }
+                ToDo newToDo = new ToDo(command.split(" ", 2)[1]);
+                this.taskList.addTask(newToDo);
+            } else if (command.startsWith("deadline")) {
+                LocalDate deadline = LocalDate.parse(command.split(" /by ", 2)[1]);
+                String name = command.split(" /by ", 2)[0].split(" ", 2)[1];
```

Comment:
Perhaps splitting them into multiple statements could improve the readability of the code. It can make it easier for you to debug your code too :)